extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TestTransactionInterceptor.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.narayana.jta.runtime.interceptor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.InvocationContext;
+import javax.transaction.UserTransaction;
+
+import io.quarkus.narayana.jta.runtime.test.TestTransactionCallback;
+
+public class TestTransactionInterceptor {
+
+    static final List<TestTransactionCallback> CALLBACKS;
+
+    static {
```

Comment:
Thanks for having considered my `flush()`-suggestion. Very cool approach with this new `TestTransactionInterceptor`!

I am wondering whether this block could kick in cases where it is not really needed? Usually `ServiceLoader` calls are not super-cheap.
I used to apply https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom but I don't know whether this would make any difference here.

Reply:
Hm, the callbacks are only loaded once per test config...

Reply:
That said, we could easily turn the list into a `io.quarkus.arc.impl.LazyValue<List<TestTransactionCallback>>` ...

Reply:
The current code is fine. ServiceLoader is expensive in terms of 'don't do this every request so you are doing it tens of thousands of times per second', its fine for this.

Reply:
Ok, thanks for your feedback.